code/add.py

```diff
@@ -18,26 +18,29 @@ def run(message, bot):
     and bot which is the telegram bot object from the main code.py function.
     """
     helper.read_json()
-    chat_id = message.chat.id
-    option.pop(chat_id, None)  # remove temp choice
     markup = types.ReplyKeyboardMarkup(one_time_keyboard=True)
     markup.row_width = 2
-    m = bot.send_message(chat_id, "Do you want to add a new category? Y/N")
-    bot.register_next_step_handler(m, post_user_def_category, bot)
+    chat_id = message.chat.id
+    for c in helper.getSpendCategories():
+        markup.add(c)
+    markup.add("Add new category")
+    msg = bot.reply_to(message, "Select Category", reply_markup=markup)
+    bot.register_next_step_handler(msg, post_category_selection, bot)
 
 
-def post_user_def_category(message, bot):
-    markup = types.ReplyKeyboardMarkup(one_time_keyboard=True)
-    markup.row_width = 2
-    chat_id = message.chat.id
-    if str(message.text) == "Y" or str(message.text) == "y":
-        message1 = bot.send_message(chat_id, "Please enter your category")
-        bot.register_next_step_handler(message1, post_append_spend, bot)
-    else:
-        for c in helper.getSpendCategories():
-            markup.add(c)
-        msg = bot.reply_to(message, "Select Category", reply_markup=markup)
-        bot.register_next_step_handler(msg, post_category_selection, bot)
+
+# def post_user_def_category(message, bot):
+#     markup = types.ReplyKeyboardMarkup(one_time_keyboard=True)
+#     markup.row_width = 2
+#     chat_id = message.chat.id
+#     if str(message.text) == "Y" or str(message.text) == "y":
+#         message1 = bot.send_message(chat_id, "Please enter your category")
+#         bot.register_next_step_handler(message1, post_append_spend, bot)
+#     else:
+#         for c in helper.getSpendCategories():
+#             markup.add(c)
+#         msg = bot.reply_to(message, "Select Category", reply_markup=markup)
+#         bot.register_next_step_handler(msg, post_category_selection, bot)
```

Comment:
Are these comments necessary?

Reply:
Thanks for catching this. Have removed them